console/client/src/providers/tabs-provider.tsx

```diff
@@ -1,33 +1,34 @@
-import { PropsWithChildren, createContext, useState } from 'react'
-import { Module, Verb } from '../protos/xyz/block/ftl/v1/console/console_pb'
+import React from 'react'
 
-export enum TabType {
-  Timeline = 'timeline',
-  Verb = 'verb',
-}
+export const TabType  = {
+  Timeline: 'timeline',
+  Verb: 'verb',
+} as const
 
 export type Tab = {
   id: string;
   label: string;
-  type: TabType;
-  module?: Module | null;
-  verb?: Verb | null;
+  type: typeof TabType[keyof typeof TabType];
 }
 
+export const TabSearchParams = {
+  active: 'active-tab',
+} as const
+
 export const timelineTab = { id: 'timeline', label: 'Timeline', type: TabType.Timeline }
 
 type TabsContextType = {
   tabs: Tab[];
-  activeTab?: Tab | null;
+  activeTab?: number;
   setTabs: React.Dispatch<React.SetStateAction<Tab[]>>;
-  setActiveTab: React.Dispatch<React.SetStateAction<Tab>>;
+  setActiveTab: React.Dispatch<React.SetStateAction<number>>;
```

Comment:
I wonder if it would be useful to add a couple more methods to this interface.
```ts
addTab(Tab)
closeTab(tab.id)
```
Then these could handle the logic for selecting the correct tab when `addTab` or `closeTab` is called

Reply:
I'm using headless and headless basses it on index/number. Honestly I prefer that to id's. It means we only have to concern ourself with keeping tab and panel order based on object position.

Reply:
I think this comment is unrelated to how headless works. This would just be for maintaining the state of what tabs we have and which are open. If `closeTab(index)` is better for you that's fine, but it would be nice for users of tabs to not have to worry about selecting the correct tab when they add or remove a tab since this `tabs-provider` can handle it for them.